pkg/operation/hybridbotanist/machines.go

```diff
@@ -213,8 +213,8 @@ func (b *HybridBotanist) generateMachineDeploymentConfig(existingMachineDeployme
 			"name":            deployment.Name,
 			"minReadySeconds": 500,
 			"rollingUpdate": map[string]interface{}{
-				"maxSurge":       1,
-				"maxUnavailable": 0,
+				"maxSurge":       deployment.MaxSurge.String(),
+				"maxUnavailable": deployment.MaxUnavailable.String(),
```

Comment:
Type changed from Int to String: Does the manifest still works?

Reply:
Yes, the parsing is to intstr.IntOrString, serialization should correctly apply here. We'll test this nevertheless.